plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBrewingStand.java

```diff
@@ -0,0 +1,147 @@
+import org.bukkit.block.data.type.BrewingStand;
+
+import java.util.Set;
+
+public class MaterialBrewingStand implements Property {
+
+    public static boolean describes(ObjectTag material) {
+        return material instanceof MaterialTag
+                && ((MaterialTag) material).hasModernData()
+                && ((MaterialTag) material).getModernData() instanceof BrewingStand;
+    }
+
+    public static MaterialBrewingStand getFrom(ObjectTag _material) {
+        if (!describes(_material)) {
+            return null;
+        }
+        else {
+            return new MaterialBrewingStand((MaterialTag) _material);
+        }
+    }
+
+    public static final String[] handledMechs = new String[] {
+            "bottles"
+    };
+
+    private MaterialBrewingStand(MaterialTag _material) {
+        material = _material;
+    }
+
+    MaterialTag material;
+
+    public static void registerTags() {
+        // <--[tag]
+        // @attribute <MaterialTag.bottles>
+        // @returns ListTag(ElementTag)
+        // @mechanism MaterialTag.bottles
+        // @group properties
+        // @description
+        // Returns a list of booleans that represent whether a slot in a brewing stand has a bottle.
+        // -->
+        PropertyParser.<MaterialBrewingStand, ListTag>registerStaticTag(ListTag.class, "bottles", (attribute, material) -> {
+            return material.getBottleBooleans();
+        });
+
+        // <--[tag]
+        // @attribute <MaterialTag.has_bottle[<#>]>
+        // @returns ElementTag(Boolean)
+        // @mechanism MaterialTag.bottles
+        // @group properties
+        // @description
+        // Returns whether the brewing stand has a bottle at the specified index.
+        // -->
+        PropertyParser.<MaterialBrewingStand, ElementTag>registerStaticTag(ElementTag.class, "has_bottle", (attribute, material) -> {
+            if (!attribute.hasParam() || !ArgumentHelper.matchesInteger(attribute.getParam())) {
+                attribute.echoError("Tag MaterialTag.has_bottle[...] must have a valid integer input.");
+                return null;
+            }
+            int index = attribute.getIntParam();
+            if (index <= 0 || index > material.getMaxBottles()) {
+                attribute.echoError("Index must be between 1 and " + material.getMaxBottles() + ".");
+                return null;
+            }
+            return new ElementTag(material.hasBottle(index-1));
+        });
+    }
+
+    public BrewingStand getBrewingStand() {
+        return (BrewingStand) material.getModernData();
+    }
+
+    public int getMaxBottles() {
+        return getBrewingStand().getMaximumBottles();
+    }
+
+    public Set<Integer> getBottles() {
+        return getBrewingStand().getBottles();
+    }
+
+    public ListTag getBottleBooleans() {
+        ListTag result = new ListTag();
+        for (int i = 0; i < getMaxBottles(); i++) {
+            result.addObject(new ElementTag(getBottles().contains(i)));

```

Comment:
This should only call `getBottles()` once, rather than for every iteration of the loop

Reply:
Actually if anything it should be calling `hasBottle`